### PR TITLE
Implement MASL for auth flow

### PR DESCRIPTION
**Pull request by ColemanMitch**
Following this tutorial here: https://docs.microsoft.com/en-us/azure/active-directory/develop/tutorial-v2-react

### CODE REVIEW

File: client/src/LogAttendance.js [hunk 1]
```diff
@@ -1,243 +1,290 @@
...
-      navigate('/responses', { state: { username: location.state.username, email }});  
-    }
-    else {
-      alert('Please enter your name, office, and practice!')
+  const handleSubmit = () => {
+    console.log(graphData);
+    if (office && practice) {
+      console.log(name, "will be in the office on these days: ", state);
+      const dbEntry = {
+        email: graphData.mail,
```
Comment (ColemanMitch) [inline client/src/LogAttendance.js:88]: the email of the authenticated account is stored within the `graphData` object

File: client/package.json [hunk 2]
```diff
@@ -12,8 +14,10 @@
     "@testing-library/react": "^12.1.4",
     "@testing-library/user-event": "^13.5.0",
     "axios": "^0.26.1",
+    "bootstrap": "^5.1.3",
     "firebase": "^9.6.10",
```
Comment (jalstadtWM) [inline client/package.json:18]: what is firebase for?

File: client/src/components/PageLayout.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import IconButton from '@mui/material/IconButton';
+
+/**
+ * Renders the navbar component with a sign-in button if a user is not authenticated
+ */
+export const PageLayout = (props) => {
+    const isAuthenticated = useIsAuthenticated();
+
+    return (
+        <>
```
Comment (jalstadtWM) [inline client/src/components/PageLayout.jsx:20]: should this have a component name?

**Review by jalstadtWM — APPROVED** [review]

File: client/package.json (hunk 2 above)
Comment (ColemanMitch) [reply client/package.json:18]: no longer being used, I'll delete it

File: client/src/components/PageLayout.jsx (hunk 3 above)
Comment (ColemanMitch) [reply client/src/components/PageLayout.jsx:20]: are you talking about the fragment (`<>`)?

**Pull request merged**